### ISSUE TITLE
[9.0] On new model install, inherit loads default call, before new xml data register

### ISSUE THREAD

**Issue by dansanti** [odoo/odoo#18562]
Impacted versions:
[9.0]

Steps to reproduce:
1.- create a new_module
2.- create a data/example.xml  with:
`
<record id="new_data" id="new.table">
<field name="name">Random Name</field>
</record>
`
3.- create an inherit class like 
`

class ResPartner(models.Model):
    _inherit ='res.partner'

    def _default_value(self):
       return self.env.ref('new_module.xml_id')

    new_field = fields.Many2one('new.table', default=lambda self: self._default_value()

class NewTable(models.Model):
    _name = 'new.table'

    name = fields.Char('New Table')
`
Current behavior:
Api loads inherit model first, include new fields from new model, 
then when call to default value, gets : 
`
ValueError: External ID not found in the system:
`
because xmlid_lookup  loads a registered module get

Expected behavior:

call "default" values before load current module

**Comment by pedrobaeza** [comment]
Yeah, this is a know problem by my part, but I'm not sure if this can be easily solved.

@rco-odoo can you please check this?

**Comment by Yenthe666** [comment]
@pedrobaeza do you happen to know if it is also known in V10 or up?

**Comment by pedrobaeza** [comment]
Yes, it's still happening

**Comment by C3POdoo** [comment]
Dear @dansanti,

Thank you for your report but we are closing it due to inactivity.
We apology if we could not look at your request in time.
If your report still makes sense, don't hesitate to reopen a new one. We will try to check it as soon as possible.

_This is an automated message._
